Code/GraphMol/ChemReactions/ReactionRunner.cpp

```diff
@@ -854,4 +882,192 @@ std::vector<MOL_SPTR_VECT> run_Reactants(const ChemicalReaction &rxn,
   return productMols;
 }  // end of ChemicalReaction::runReactants()
 
+// Generate a product based on a SINGLE reactant
```

Comment:
It's worse, it is a vector of vector of mols, each reaction may have multiple product templates and each product template may have multiple "products".

It's kind of mind bending, nevertheless the product in the comment should be plural.